testing/mod.ts

```diff
@@ -73,11 +73,20 @@ export const exitOnFail = true;
 
 let filterRegExp: RegExp | null;
 const tests: TestDefinition[] = [];
+const setUps: TestFunction[] = [];
+const tearDowns: TestFunction[] = [];
 
 let filtered = 0;
 const ignored = 0;
 const measured = 0;
 
+export function setUp(t: TestFunction): void {
+  setUps.push(t);
+}
+export function tearDown(t: TestFunction): void {
+  tearDowns.push(t);
+}
```

Comment:
Can we just fix timer for starting fileServer in this PR and leave out the setUP and tearDown stuff? 

Because modifying testing will effect all of deno, this needs to be considered very carefully. I'm not sure I agree with setting a global setUp/tearDown. 

Please just manually do the setUP and tearDown for each of the fileServer tests with the new initialization. After that's landed let's discuss these testing changes (in gitter if you have time)

Reply:
>  I'm not sure I agree with setting a global setUp/tearDown.

I would prefer this was per-file, but am not sure on the trick to make this work.

I also want to improve this across all of deno / have it considered carefully.

I could tease it apart:
- timing of fileserver
- main.ts for testing
- setUp + tearDown (to discuss)

Reply:
> timing of fileserver

Let's do just this in this PR. 

> main.ts for testing

Yeah. I think we should get rid of the "automatic" run behavior that we currently have 
https://github.com/denoland/deno_std/blob/b99d7d3e0f61216db7b5a5ad2bc254f8ac97ff4d/testing/mod.ts#L166
and rather require users to call runTests() explicitly.

Reply:
@hayd I'm very interested in having your improvements here. I do think they should be split up into different PRs tho. For simplicity, I'm going to close this one.